src/main/java/eiteam/esteemedinnovation/commons/recipe/ConfigConditionFactory.java

```diff
@@ -0,0 +1,24 @@
+package eiteam.esteemedinnovation.commons.recipe;
+
+import com.google.gson.JsonObject;
+import eiteam.esteemedinnovation.commons.Config;
+import net.minecraft.util.JsonUtils;
+import net.minecraftforge.common.crafting.IConditionFactory;
+import net.minecraftforge.common.crafting.JsonContext;
+
+import java.util.function.BooleanSupplier;
+
+public class ConfigConditionFactory implements IConditionFactory {
+
+    @Override
+    public BooleanSupplier parse(JsonContext context, JsonObject json) {
+        String configSetting = JsonUtils.getString(json, "config_setting", "");
+
+        switch (configSetting) {
+            //TODO: add config options here
+            case "enableAstrolabe": return () -> Config.enableAstrolabe;
```

Comment:
I think if we are going to be overhauling the way configuration works, we should put config options into `ContentModule`s. Do you have any ideas for how we could do that? I've never heard of this `ConditionFactory` stuff.

Reply:
Since moving to json recipes, we need a way to enable/disable certain recipes depending on config settings. Forge has some conditions built in for if a certain mod is loaded, etc... `ConfigConditionFactory` allows us to have separate config options to disable each individual recipe. (As the old code had)

Reply:
I will mess around with this after merged. I'm fine with this for the purposes of this PR, will probably split it up into the ContentModules before first release though.